src/thx/Ord.hx

```diff
@@ -2,15 +2,34 @@ package thx;
 
 import thx.Semigroup;
 
-enum Ordering {
+abstract Ordering(OrderingImpl) from OrderingImpl to OrderingImpl {
+  @:from public static function fromInt(value : Int) : Ordering
+    return value < 0 ? LT : (value > 0 ? GT : EQ);
+
+  public static function fromFloat(value : Float) : Ordering
+    return value < 0 ? LT : (value > 0 ? GT : EQ);
+
+  @:to public function toInt() : Int
+    return switch this {
+      case LT: -1;
+      case GT: 1;
+      case EQ: 0;
+    };
+}
+
+enum OrderingImpl {
   LT; GT; EQ;
 }
 
+@:callable
 abstract Ord<A> (A -> A -> Ordering) from A -> A -> Ordering to A -> A -> Ordering {
-  public function order(a0: A, a1: A): Ordering
+  @:from public static function fromIntComparison<A>(f : A -> A -> Int) : Ord<A>
```

Comment:
Same objection here, I don't like that this can be implicitly created from any function of this shape. Explicit is always clear, implicit means that the compiler may admit things that you don't want.